src/main/java/org/opensearch/performanceanalyzer/decisionmaker/actions/SearchBackPressureAction.java

```diff
@@ -0,0 +1,305 @@
+import org.opensearch.performanceanalyzer.AppContext;
+import org.opensearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+
+public class SearchBackPressureAction extends SuppressibleAction {
+    private static final Logger LOG = LogManager.getLogger(SearchBackPressureAction.class);
+    public static final String NAME = "SearchBackPressureAction";
+    private static final ImpactVector NO_IMPACT = new ImpactVector();
+
+    /*
+     * Time to wait since last recommendation, before suggesting this action again
+     */
+    private static final long DEFAULT_COOL_OFF_PERIOD_IN_MILLIS = TimeUnit.DAYS.toMillis(1);
+
+    /* From Config Per Diumension Type
+     *  canUpdate: whether the action should be emitted
+     *  coolOffPeriodInMillis: how long the CoolOffPeriod the action should before reemit
+     *  thresholdName: the name of threshold we are tuning  (e.g. node_duress.cpu_threshold, search_heap_threshold)
+     *  dimension: indicates whether the resource unit is caused by shard/task level searchbackpressure cancellation stats
+     *  Step Size in percentage: how much should the threshold change in percentage
+     */
+    private boolean canUpdate;
+    private long coolOffPeriodInMillis;
+    private String thresholdName;
+
+    // TODO: change dimension, direction as enum
+    private SearchbpDimension dimension;
+    private SearchbpThresholdActionDirection direction;
+    private double stepSizeInPercentage;
+
+    public SearchBackPressureAction(
+            final AppContext appContext,
+            final boolean canUpdate,
+            final long coolOffPeriodInMillis,
+            final String thresholdName,
+            final SearchbpDimension dimension,
+            final SearchbpThresholdActionDirection direction,
+            final double stepSizeInPercentage) {
+        super(appContext);
+        this.canUpdate = canUpdate;
+        this.coolOffPeriodInMillis = coolOffPeriodInMillis;
+        this.thresholdName = thresholdName;
+        this.dimension = dimension;
+        this.direction = direction;
+        this.stepSizeInPercentage = stepSizeInPercentage;
+    }
+
+    @Override
+    public String name() {
+        return NAME;
+    }
+
+    @Override
+    public boolean canUpdate() {
+        return canUpdate;
+    }
+
+    @Override
+    public long coolOffPeriodInMillis() {
+        return coolOffPeriodInMillis;
+    }
+
+    @Override
+    public List<NodeKey> impactedNodes() {
+        // all nodes are impacted by this change
+        return appContext.getDataNodeInstances().stream()
+                .map(NodeKey::new)
+                .collect(Collectors.toList());
+    }
+
+    /* TO DO: Discuss the impact of SearchBackPressureAction
+     * since our action only modify the threhsold settings of Search Back Pressure Service instead of actual Resource
+     * No Impact should be put as the Impact Vector for this action so other actions would not be affected by Searchbp-specific actions
+     */
```

Comment:
Can you elaborate more on this ?
What is the threshold we are tuning in regard to Search BackPressure and how does it co-relate to Dimensions mentioned at [ImpactVector.java](https://github.com/opensearch-project/performance-analyzer-rca/blob/main/src/main/java/org/opensearch/performanceanalyzer/decisionmaker/actions/ImpactVector.java#L16)?

Reply:
+1

Reply:
3things to clarify for Impact Vector and Search Back Pressure (Searchbp) Auto-tuning.

1. Search Back Pressure Decider/Policy only tunes searchbackpressure related thresholds (e.g. search_backpressure.search_task_heap_threshold) and it does not correlate directly with any current dimension in the ImpactVector (e.g. CPU/HEAP).
2. Current Searchbp actions only adjust heap related Searchbp Thresholds.
3. Dimensions in ImpactVector is used by collator to determine which action should be emitted to Publisher, eventually which actions should the downstream class execute. So if there are 2 actions emitting in the same time, one increase CPU and one decrease it, the collator cancel out the actions. However, since for Searchbp Actions we only tune the searchbp threshold once per threshold (it's impossible for 2 actions emitting in the same time that increase and decrease searchbackpressure heap usage threshold). Therefore, we put no Impact for ImpactVector for Searchbp Actions.
```
@Override
    public Map<NodeKey, ImpactVector> impact() {
        Map<NodeKey, ImpactVector> impact = new HashMap<>();
        for (NodeKey key : impactedNodes()) {
            impact.put(key, NO_IMPACT);
        }
        return impact;
    }
```